### PR TITLE
[5.2] foreach usage

### PR DESCRIPTION
**Pull request by Denitz**
### Summary of Changes

Prefer foreach() over for(), especially for iterating objects. Foreach is simpler.

### Testing Instructions

Apply patch.

### Actual result BEFORE applying this Pull Request

Complex code.

### Expected result AFTER applying this Pull Request

Simple code.

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Comment by HLeithner** [comment]
This pull request has been automatically rebased to 5.1-dev.

File: administrator/components/com_joomlaupdate/src/Model/UpdateModel.php [hunk 1]
```diff
@@ -1298,8 +1298,8 @@ public function getIniParserAvailability()
             $disabledFunctions         = explode(',', trim($disabledFunctions));
             $numberOfDisabledFunctions = \count($disabledFunctions);
```
Comment (Quy) [inline administrator/components/com_joomlaupdate/src/Model/UpdateModel.php]:
```suggestion
```

File: administrator/components/com_login/src/Model/LoginModel.php [hunk 2]
```diff
@@ -82,10 +82,10 @@ public static function getLoginModule($name = 'mod_login', $title = null)
         $modules = self::_load($name);
         $total   = \count($modules);
```
Comment (Quy) [inline administrator/components/com_login/src/Model/LoginModel.php]:
```suggestion
```

File: administrator/components/com_modules/src/Controller/ModuleController.php [hunk 3]
```diff
@@ -294,16 +294,16 @@ public function orderPosition()
         $n       = \count($orders);
 
         if ($n > 0) {
```
Comment (Quy) [inline administrator/components/com_modules/src/Controller/ModuleController.php]:
```suggestion

        if (\count($orders)) {
```

File: libraries/src/HTML/Helpers/StringHelper.php [hunk 4]
```diff
@@ -107,11 +107,11 @@ public static function truncate($text, $length = 0, $noSplit = true, $allowHtml
                     $openedTags = array_reverse($openedTags);
```
Comment (Quy) [inline libraries/src/HTML/Helpers/StringHelper.php:107]: No need for assignment on line 102 and should be used directly on line 105.

**Comment by rdeutz** [comment]
This needs to be done carefully because a foreach makes a copy of the array and works with the copy so a foreach consume more memory. So if you've a large array you can hit the memory_limit, if it is set to a low value.

**Comment by HLeithner** [comment]
This pull request has been automatically rebased to 5.2-dev.

**Review by Hackwar — APPROVED** [review]

**Review by HLeithner — DISMISSED** [review]

File: administrator/components/com_users/src/Model/LevelModel.php [hunk 5]
```diff
@@ -300,11 +300,11 @@ public function validate($form, $data, $group = null)
...
-            for ($i = 0, $n = \count($groups); $i < $n; ++$i) {
-                if (Access::checkGroup((int) $groups[$i]->id, 'core.admin')) {
-                    if (\in_array((int) $groups[$i]->id, $rules) && !\in_array((int) $groups[$i]->id, $data['rules'])) {
-                        $data['rules'][] = (int) $groups[$i]->id;
-                    } elseif (!\in_array((int) $groups[$i]->id, $rules) && \in_array((int) $groups[$i]->id, $data['rules'])) {
+            foreach ($groups as $value) {
+                if (Access::checkGroup((int) $value->id, 'core.admin')) {
+                    if (in_array((int) $value->id, $rules) && !in_array((int) $value->id, $data['rules'])) {
+                        $data['rules'][] = (int) $value->id;
+                    } elseif (!in_array((int) $value->id, $rules) && in_array((int) $value->id, $data['rules'])) {
```
Comment (richard67) [inline administrator/components/com_users/src/Model/LevelModel.php]:
```suggestion
                    if (\in_array((int) $value->id, $rules) && !\in_array((int) $value->id, $data['rules'])) {
                        $data['rules'][] = (int) $value->id;
                    } elseif (!\in_array((int) $value->id, $rules) && \in_array((int) $value->id, $data['rules'])) {
```
Fix PHPCS (1)

File: libraries/src/HTML/Helpers/StringHelper.php [hunk 6]
```diff
@@ -101,19 +101,17 @@ public static function truncate($text, $length = 0, $noSplit = true, $allowHtml
...
+                if (\count($closedTags) !== \count($openedTags)) {
                     // Closing tags need to be in the reverse order of opening tags.
                     $openedTags = array_reverse($openedTags);
 
                     // Close tags
-                    for ($i = 0; $i < $numOpened; $i++) {
-                        if (!\in_array($openedTags[$i], $closedTags)) {
-                            $tmp .= '</' . $openedTags[$i] . '>';
+                    foreach ($openedTags as $openedTag) {
+                        if (!in_array($openedTag, $closedTags)) {
```
Comment (richard67) [inline libraries/src/HTML/Helpers/StringHelper.php]:
```suggestion
                        if (!\in_array($openedTag, $closedTags)) {
```
Fix PHPCS (2 of 2)

**Review by laoneo — APPROVED** [review]

**Review by MacJoom — APPROVED** [review]

**Comment by Hackwar** [comment]
Thank you for your contribution @Denitz!

**Pull request merged**